ecleankernel/layout/blspec.py

```diff
@@ -149,6 +151,17 @@ def find_kernels(self,
                         ver, module_dict,
                         exclusions)
 
+                # Debian's installkernel also installs config and System.map
```

Comment:
Yes, with debian's instalkernel (USE=-systemd), USE=efistub and USE=uki relocate the `/boot` install to the ESP (`ESP/EFI/Gentoo` for efistub and `ESP/EFI/Linux` for uki) including the config and system.map. 

Systemd `kernel-install` does not do this since it never considers the config or system.map in the first place.

You could argue that installing the config is not really necessary in the first place (and I'd mostly agree), but I know some folks prefer debian's installkernel because they like that it installs the config. And if we must install some config then it only makes sense to install it on the same partition as the kernel and in the same directory. If the config would be on some other partition then this would kind of defeat the purpose of installing the config in the first place.

Reply:
Doesn't this mess up UEFI bootloaders, though? I.e. cause them to consider the file as another kernel.

Reply:
No, UEFI has no auto-discovery, it only considers whatever you explicitly add a boot entry for (and it often, but not always, enforces that the file being booted via that entry has the `.efi` suffix).

Systemd-boot and rEFInd do have auto-discovery but they discover only files that have the `.efi` suffix, the exception being that rEFInd also looks for an initramfs and/or icon with the same name as the discovered kernel but a different suffix/prefix.

Reply:
Still, it feels that you're hijacking the UKI layout with files that don't belong there, and don't use consistent naming with the spec.

Reply:
Well debian's installkernel never followed the spec exactly, it should install the uki's with the 'entry token' as the name, but the concept of the entry token is specific to systemd so we instead simply use the ID from os-release. And rEFInd also does not follow the spec exactly since we need an icon file in the UKI directory which according to the spec should not be there.

It's nice that there is a spec and I do try to follow it as much as possible, but the reality is that the only bootloader that cares about the spec is systemd-boot, and when we use systemd-boot then we also use systemd's kernel-install which does follow the spec exactly. At some point we have to make some compromise between real functionality (e.g. UKI support for rEFInd) and the spec (which does not recognize the concept of icon files).

I can remove the config and system.map files from the UKI dir, but I am pretty sure this will upset some people. Instead I think that users who care about systemd's spec should just use systemd's tool (i.e. kernel-install). This is already enforced when systemd-boot is used.

Reply:
Are we really talking about Debian, or are we talking about our fork of installkernel that has nothing to do with the original anymore?

Reply:
Our fork. I still call that debian installkernel to differentiate from systemd kernel-install.

Reply:
Actually, at this point I'm wondering if there's a point in having two separate layouts if blspec layout is effectively a junk directory like the std layout.

Reply:
> Actually, at this point I'm wondering if there's a point in having two separate layouts if blspec layout is effectively a junk directory like the std layout.

What would be the alternative? Handeling all possible boot-files/locations in a single python file?

There's still a bunch of differences between `std.py` and `blsspec.py`. For example the UKI directory will always contain UKIs, whereas `/boot` or the efistub directory always contain regular kernel images. And then there is the systemd-boot type 1 directory, which is very distinct from the other possible locations in its structure.

Reply:
Dropped the config/system-map part, will adjust installkernel itself later.

Reply:
> Dropped the config/system-map part, will adjust installkernel itself later.

Resolved now in installkernel v38.